test/unit/governance/LendingTermOnboarding.t.sol

```diff
@@ -0,0 +1,297 @@
+        onboarder.proposeOnboard(address(term));
+
+        // mint GUILD & self delegate
+        guild.mint(alice, _PROPOSAL_THRESHOLD);
+        guild.mint(bob, _QUORUM);
+        vm.prank(alice);
+        guild.delegate(alice);
+        vm.prank(bob);
+        guild.incrementDelegation(bob, _QUORUM);
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 13);
+
+        // propose onboard
+        vm.prank(alice);
+        uint256 proposalId = onboarder.proposeOnboard(address(term));
+
+        // cannot propose the same term multiple times in a short interval of time
+        vm.expectRevert("LendingTermOnboarding: recently proposed");
+        onboarder.proposeOnboard(address(term));
+
+        // check proposal creation
+        assertEq(uint8(onboarder.state(proposalId)), uint8(IGovernor.ProposalState.Pending));
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 13);
+        assertEq(uint8(onboarder.state(proposalId)), uint8(IGovernor.ProposalState.Active));
+
+        // support & check status
+        vm.prank(bob);
+        onboarder.castVote(proposalId, uint8(GovernorCountingSimple.VoteType.For));
+        vm.roll(block.number + _VOTING_PERIOD + 1);
+        vm.warp(block.timestamp + 13);
+        assertEq(uint8(onboarder.state(proposalId)), uint8(IGovernor.ProposalState.Succeeded));
+
+        // queue
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 13);
+        (
+            address[] memory targets,
+            uint256[] memory values,
+            bytes[] memory calldatas,
+            string memory description
+        ) = onboarder.getOnboardProposeArgs(address(term));
+        onboarder.queue(targets, values, calldatas, keccak256(bytes(description)));
+        assertEq(uint8(onboarder.state(proposalId)), uint8(IGovernor.ProposalState.Queued));
+
+        // execute
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + _TIMELOCK_MIN_DELAY + 13);
+        onboarder.execute(targets, values, calldatas, keccak256(bytes(description)));
+        assertEq(uint8(onboarder.state(proposalId)), uint8(IGovernor.ProposalState.Executed));
+
+        // check execution
+        assertEq(guild.isGauge(address(term)), true);
+
+        // cannot propose the same term twice if it's already active
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 7 days + 13);
+        vm.expectRevert("LendingTermOnboarding: active term");
+        onboarder.proposeOnboard(address(term));
+
+        // vote for the term
+        guild.mint(address(this), 1e18);
+        guild.incrementGauge(address(term), 1e18);
+
+        // do a borrow
+        collateral.mint(address(this), 1e24);
+        collateral.approve(address(term), 1e24);
+        term.borrow(1e24, 1e24);
+        assertEq(collateral.balanceOf(address(this)), 0);
+        assertEq(collateral.balanceOf(address(term)), 1e24);
+        assertEq(credit.balanceOf(address(this)), 1e24);
+    }
```

Comment:
I'd like to see two tests:

if gauge was active, then was offboarded, it can be added again through the onboarding contract

if gauge was active, then was deprecated, it can be added again through the onboarding contract

Reply:
sounds like an integration test between `LendingTermOnboarding` & `LendingTermOffboarding` 😉

Reply:
The onboarding is queued in the timelock so I cannot add "at runtime checks", but I'll add these checks in the lending term OFFboarding

Reply:
The onboarding is queued in the timelock so I cannot add "at runtime checks", but I'll add these checks in the lending term OFFboarding

Reply:
done